papers/whitepaper.md

```diff
@@ -111,3 +113,141 @@ The focus of the current text is to explore at a deeper level the Atoma's unique
+
+Nodes on the Atoma Network listen to blockchain events that are emitted once new request transactions are submitted to the Atoma on-chain logic contract. Once it listens to a new event, it can check if it has been selected to process the request, or not. If it has been selected, it should start processing the request right away. If the event contains all the available input data (in case, the transaction was submitted another smart contract in a `Smart contract authenticated request` case), or it must then fetch the input data from an external API call (in case, the transaction was submitted through a `Externally owned request` case) to the application the user is interacted with, so that it can start processing the request, right away.
+
+**Node collateral and rewards:** As already mentioned, nodes on the Atoma Network are required to deposit a given amount of collateral, upon registration. This collateral is subject to slashing in the cases of malicious behavior or in case the node becomes unresponsive (it stops processing requests for a long period of time). It was already mentioned that nodes are only able to withdraw that collateral in the epoch following de-registration. 
+
+If a node timeouts for a given request, a fixed percentage of its deposited collateral is slashed. Part of the slashed collateral goes to the user that initiated the request, to compensate the users for a poor service. Another part of the slashed collateral goes to the Atoma community treasury fund, and the remaining part is burned. This incentivizes nodes to be always responsive, otherwise they lose part of their stake in the network, while compensating both the user (with a refund), the network through treasury fund allocation and contributing for the TOMA token appreciation through deflation.
+
+On the other hand, if a node acts maliciously, it gets penalized with all of its deposited collateral being slashed. Similarly to the timeout case above, a percentage of the slashed collateral goes to the user, compensating for any possible malicious behavior. There is also a percentage of the slashed collateral that is allocated to the honest nodes involved in processing the request, the Atoma nodes that participated in the dispute event, to the Atoma community treasury fund, and the remaining collateral is burned.
+
+On the other hand, for each processed request, a node is entitled to part of the fee paid by the user that initiated the request. Request fees are paid in `TOMA` tokens and their price is set by a fair free market mechanism. At the end of each epoch, the Atoma nodes bid the TOMA token fee value, per token, for the next epoch. Therefore, fees are computed in terms of the total number of tokens a given request requires to be processed. So for example, if a prompt text request has 100 input tokens, and its output response contains 200 tokens, the total fee amount paid to a node to process that request will be none other than:
+
```

Comment:
Last line: 

The total fee amount paid to a node to process request will be calculated as: